File: algorithms/medium/smallest_string_with_a_given_numeric_value.ex
defmodule Solution do
  @spec get_smallest_string(n :: integer, k :: integer) :: String.t()
  def get_smallest_string(n, k) do
    aux(n, k)
  end

  def aux(n, k), do: aux(n, k, [])
  def aux(0, _k, s), do: List.to_string(s)

  def aux(n, k, s) do
    c = min(26, k - n + 1)
    aux(n - 1, k - c, [97 + c - 1 | s])
  end
end

File: algorithms/medium/smallest_string_with_a_given_numeric_value.rs
impl Solution {
    pub fn get_smallest_string(mut n: i32, mut k: i32) -> String {
        let mut s = String::new();        
        while n > 0 {
            let c = 26.min(k - (n - 1));
            s.push((b'a' + c as u8 - 1) as char);
            n -= 1;
            k -= c;
        }
        s.chars().rev().collect()
    }
}
